### ISSUE TITLE
fix: shorebird reports 3.7.10 even though we're on 3.7.12

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#364]
I'm not sure what's wrong.

Felix found the logic in the `flutter` tool:
```dart
// If we're not currently on a tag, use git describe to find the most
// recent tag and number of commits past.
return parse(
  _runGit(
    'git describe --match *.*.* --long --tags $gitRef',
    processUtils,
    workingDirectory,
  )
);
```

**Comment by eseidel** [comment]
https://github.com/flutter/flutter/blob/932faedce2f8277c1190e981367662d2799899bf/dev/conductor/core/lib/src/version.dart#L28

**Comment by eseidel** [comment]
We've disabled the Flutter validator on most commands so this shouldn't show up very often:
https://github.com/shorebirdtech/shorebird/issues/341

It may just fix itself as part of our next rebase of Flutter.

**Comment by eseidel** [comment]
This seems to be happening again with our latest fork.  Says 3.10.0-pre rather than just 3.10.0.

**Comment by bryanoltman** [comment]
This has evolved to:

```
⑆ ./bin/flutter --version
Flutter 3.9.0-20.0.pre.209 • channel unknown • unknown source
Framework • revision c0e52af909 (12 days ago) • 2023-05-05 13:53:49 -0500
Engine • revision 13b3ad54e8
Tools • Dart 3.0.0 • DevTools 2.23.1
```
despite being on Flutter 3.10.0.

This is causing problems when checking version constraints on dependencies.

From Nils in the support channel:

> We can't publish release and patches for Shorebird since Flutter 3.10 anymore because of https://github.com/shorebirdtech/shorebird/issues/364. When building the Android with Shorebird (shorebird release --flavor prodAlpha --target lib/main_prod.dart), I'm getting:
> 
> ✗ Failed to build: Flutter assets will be downloaded from https://download.shorebird.dev. Make sure you trust this source!
> The current Flutter SDK version is 3.9.0-20.0.pre.209.
> 
> Because screenshot 2.1.0 requires Flutter SDK version >=3.10.0 and no versions of screenshot match >2.1.0 <3.0.0, screenshot ^2.1.0 is forbidden.
> So, because familymissions depends on screenshot ^2.1.0, version solving failed.
> 
> 
> The package screenshot has flutter: '>=3.10.0' has environment constrain: https://github.com/SachinGanesh/screenshot/blob/ae826081dc7e026b19763f2a53fac9bce956e15c/pubspec.yaml#L8
>

**Comment by bryanoltman** [comment]
The `git describe` command (`git describe --match *.*.* --long --tags c0e52af9097e779671591ea105031920f24da4d5`) is saying that no matches are found. However, `bin/cache/flutter/version` contains `3.9.0-20.0.pre.209`. Maybe we just update this?

**Comment by bryanoltman** [comment]
It seems like `version` gets written out as part of `FLUTTER_STORAGE_BASE_URL=https://download.shorebird.dev ./bin/flutter --version`. Digging further…

**Comment by eseidel** [comment]
I expect this will mostly be about understanding how the flutter code works and then figuring out what we're doing wrong with our tagging (or tag fetching?) in our checkout that's confusing the code.  It's possible we're just not fetching all teh tags we should or that maybe we're constructing our forked branch incorrectly and missing teh last commit with the tag?

**Comment by bryanoltman** [comment]
Digging into the Flutter version code now, trying to figure out why `git describe --match *.*.* --long --tags HEAD` prints something different when run by the flutter tool than it does when I run it in my shell 🤔

**Comment by bryanoltman** [comment]
Fix: the match regex needs quotes around it to work, at least on zsh on macOS. `git describe --match "*.*.*" --long --tags HEAD` prints `3.9.0-19.0.pre-209-gc0e52af90`

**Comment by bryanoltman** [comment]
Ok, this is fixed. The issue was that we were not syncing our [vended Flutter](https://github.com/shorebirdtech/flutter)'s tags with the [upstream Flutter](https://github.com/flutter/flutter)'s. The steps to fix this were added to docs in eaa1ef45c33c4f74a41aa9314ab5f3f64a4efab8, although this still reports an incorrect version, albeit a less incorrect one.

**Comment by bryanoltman** [comment]
Filed https://github.com/shorebirdtech/shorebird/issues/508 to track the x.y.z+1 issue.
